### PR TITLE
Replaced fire with Argparse

### PR DESCRIPTION
**Pull request by Flodur871**
I replaced fire with Argparse as discussed in #38. 
Additionally I added some missing docstrings and made small fixes to the code.

### CODE REVIEW

**Comment by Adwaith-Rajesh** [comment]
@Flodur871 thanks for the contribution, You PR will be merged by tomorrow.

File: pysondb/cli.py [hunk 1]
```diff
@@ -135,18 +147,72 @@ def verify_file(file_data, refer_keys, filename):
...
-        {
-            "create": create_if_not_exist,
-            "display": display,
-            "delete": delete,
-            "convert": convert,
-            "converttocsv": convert_db_to_csv,
-            "merge": merge,
-        }
+def set_parser():
+    parser = argparse.ArgumentParser()
```
Comment (Adwaith-Rajesh) [inline pysondb/cli.py:151]: The `ArgumentParser` should now accept `argv` as an argument
`argparse.ArgumentParser(argv)`

File: pysondb/cli.py [hunk 2]
```diff
@@ -135,18 +147,72 @@ def verify_file(file_data, refer_keys, filename):
...
+        convert_db_to_csv(args.file_name, args.target_csv)
+
+    elif args.command == "merge":
+        merge(args.p_file, args.m_file, args.output_file)
+
+    else:  # show help menu if the cli was started without an argument
+        parser.print_help()
+
+
+def main():
```
Comment (Adwaith-Rajesh) [inline pysondb/cli.py:213]: `def main() -> int`

File: pysondb/cli.py [hunk 3]
```diff
@@ -135,18 +147,72 @@ def verify_file(file_data, refer_keys, filename):
...
+    set_parser()
+
 
 if __name__ == "__main__":
     main()
```
Comment (Adwaith-Rajesh) [inline pysondb/cli.py:218]: `raise SystemExit(main())` So that the exit codes for that last command is stored in the terminal session

**Review by Adwaith-Rajesh — CHANGES_REQUESTED** [review]
you can ignore the type hints if you want to, but the rest of the thing are needed for the long run

**Review by fredysomy — CHANGES_REQUESTED** [review]
hey @Flodur871 It will be great if you can update the [docs](https://fredysomy.me/pysonDB/docs/cli) with the updated arguments.

**Comment by fredysomy** [comment]
do you guys find any alternatives to beautifultable

**Comment by Adwaith-Rajesh** [comment]
> do you guys find any alternatives to beautifultable

https://stackoverflow.com/a/26937531 or we have to use python string formatting

**Comment by fredysomy** [comment]
I was thinking of reducing some more depency size. And at the same time
display method should be there as a visual feedback for the developer.

Is it complicated using format strings

**Comment by Flodur871** [comment]
> hey @Flodur871 It will be great if you can update the [docs](https://fredysomy.me/pysonDB/docs/cli) with the updated arguments.

Will do

**Review by Adwaith-Rajesh — APPROVED** [review]
nice work, thnx for the contributing.

**Review by fredysomy — APPROVED** [review]

**Comment by fredysomy** [comment]
Thanks for the Contribution.🤩

**Pull request merged**